### ISSUE TITLE
SSL not enabled before connected

### ISSUE THREAD

**Issue by puschie286** [DirectoryTree/LdapRecord#57]
- LDAP Server Type: AD
- PHP Version: 7.3

### Description:
SSL (and TLS) are not set before ldap_connect is called
-> ErrorCode: -1, Message: Can't contact LDAP 
( it tries to connect with ldap and not with ldaps )
### Steps To Reproduce:
configure AD server connection
enable ssl

### Workaround:
move setting ssl (and tls) from Connection::configure to Connection::initialize before Connection::connect is called

**Comment by stevebauman** [comment]
Hi @puschie286, are you able to post your code?

Are you sure your LDAP server is reachable?

**Comment by puschie286** [comment]
yes - everything works fine except it tries to connect with ldap:// instead of ldaps:// which doesnt work because of the port

my workaround code:
```
public function initialize()
{
	if ($this->configuration->get('use_ssl')) {
		$this->ldap->ssl();
	} elseif ($this->configuration->get('use_tls')) {
		$this->ldap->tls();
	}

	$this->ldap->connect(
            $this->configuration->get('hosts'),
            $this->configuration->get('port')
        );

        $this->configure();
}
```

**Comment by stevebauman** [comment]
The LDAPS protocol is used when defining the connection string here:

https://github.com/DirectoryTree/LdapRecord/blob/e0eeba7777241a2d8bc4cfc015d712b9a2d5a6be/src/Ldap.php#L820

And then used when building the connection string:

https://github.com/DirectoryTree/LdapRecord/blob/e0eeba7777241a2d8bc4cfc015d712b9a2d5a6be/src/Ldap.php#L408

**Comment by stevebauman** [comment]
The LDAPS protocol used when defining the connection string here:

https://github.com/DirectoryTree/LdapRecord/blob/e0eeba7777241a2d8bc4cfc015d712b9a2d5a6be/src/Ldap.php#L820

**Comment by stevebauman** [comment]
Sorry - fat fingers and accidentally closed the issue. I see what you’re saying, thanks for pointing that out!

I’ll patch this shortly.

**Comment by stevebauman** [comment]
Fixed - ccc18b9

Thanks for the report :smile:
